src/wp-admin/includes/image.php

```diff
@@ -449,24 +571,168 @@ function _wp_make_subsizes( $new_sizes, $file, $image_meta, $attachment_id ) {
 		}
 	}
 
+	// Filter supported sizes for non primary mime types.
+	if ( ! $is_primary_mime ) {
+		$supported_multi_mime_sizes = _wp_multi_mime_get_supported_sizes( $attachment_id );
+		$new_sizes                  = array_filter(
+			$new_sizes,
+			function( $size ) use ( $supported_multi_mime_sizes ) {
+				return in_array( $size, $supported_multi_mime_sizes );
+			},
+			ARRAY_FILTER_USE_KEY
+		);
+	}
```

Comment:
Reviewing this more closely, I don't think that makes sense here. We have to make sure our code works correctly both for a (likely theoretical) upload where all files are generated right away and for an upload where multiple requests are needed (see `wp_get_missing_image_subsizes()`).

We should move this outside of this function, everywhere that `_wp_make_subsizes()` may be called for an additional MIME type, the sizes to generate should then be filtered. This is most critical to move outside of here for the `wp_get_missing_image_subsizes()` use-case because otherwise it'll continue to think there are missing sub-sizes for a MIME type although the missing sub-sizes may not be relevant for that MIME type.

To reduce the code duplication from that, we may want to change the function `_wp_multi_mime_get_supported_sizes( $attachment_id )` to be more of a "filter" function that filters a given `$sizes` array using the array from the filter. I don't think we need to just "get" that list anywhere, so we could just replace that function. If we both need to get it _and_ use it to filter, we could just create both functions.

Reply:
* adjusted function (now named `_wp_filter_image_sizes_additional_mime_type_support`) to accept a `$sizes` array and return a filtered sizes array.
* calling this function to filter missing sizes in `wp_get_missing_image_subsizes` (non-primary mime only) - this is called in `wp_update_image_subsizes` to get the missing sizes, so no need to update there.
* call in `wp_create_image_subsizes` before handling the secondary mime type(s)